### PR TITLE
Fix OOM issue in Serverless

### PR DESCRIPTION
**Pull request by michal-hudy**
**Description**

Changes proposed in this pull request:

- Fixed OOM crash loop issue
- Fixed pod crash because of `fatal error: concurrent map writes`
- Simplified secrets/runtimes propagation
- Fixed issue with missing `ConfigurationReady` condition

I will add missing Unit Tests in a follow-up PR.

**Related issue(s)**

#8237

### CODE REVIEW

File: components/function-controller/README.md [hunk 1]
```diff
@@ -23,10 +23,11 @@ The Function Controller uses these environment variables:
...
+| **APP_KUBERNETES_SECRET_REQUEUE_DURATION** | Period of time after which the Secret Controller refreshes the status of a Secret                                                                                                                                           | `1m`                                                                                                                                                       |
+| **APP_KUBERNETES_SERVICE_ACCOUNT_REQUEUE_DURATION** | Period of time after which the ServiceAccount Controller refreshes the status of a ServiceAccount                                                                                                                                           | `1m`                                                                                                                                                       |
```
Comment (kazydek) [inline components/function-controller/README.md:30]:
```suggestion
| **APP_KUBERNETES_SERVICE_ACCOUNT_REQUEUE_DURATION** | Period of time after which the Service Account Controller refreshes the status of a service account                                                                                                                                         | `1m`                                                                                                                                                       |
```

File: components/function-controller/internal/controllers/kubernetes/serviceaccount_service.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+	if err := r.client.Update(ctx, copy); err != nil {
+		logger.Error(err, fmt.Sprintf("Updating ServiceAccount '%s/%s' failed", copy.GetNamespace(), copy.GetName()))
+		return err
+	}
+
+	return nil
+}
+
+func (*serviceAccountService) shiftSecretTokens(baseInstance *corev1.ServiceAccount) []corev1.ObjectReference {
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply components/function-controller/internal/controllers/kubernetes/serviceaccount_service.go:105]: I wonder if there will ever be duplication of secrets using these two functions.

File: components/function-controller/config/rbac/role.yaml [hunk 3]
```diff
@@ -32,6 +32,18 @@ rules:
   - jobs/status
   verbs:
   - get
+- apiGroups:
```
[the remark this answers is not in the document]
Comment (michal-hudy) [reply components/function-controller/config/rbac/role.yaml:35]: It is generated...

File: components/function-controller/internal/controllers/kubernetes/secret_service.go [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+	return nil
+}
+
+func (r *secretService) updateSecret(ctx context.Context, logger logr.Logger, instance, baseInstance *corev1.Secret) error {
+	copy := instance.DeepCopy()
+	copy.Annotations = baseInstance.GetAnnotations()
+	copy.Labels = baseInstance.GetLabels()
+	copy.Data = baseInstance.Data
+	copy.StringData = baseInstance.StringData
+	copy.Type = baseInstance.Type
```
[the remark this answers is not in the document]
Comment (michal-hudy) [reply components/function-controller/internal/controllers/kubernetes/secret_service.go:89]: OK, leave as it is for now, as we will not change type on release 1.12

File: components/function-controller/internal/controllers/kubernetes/serviceaccount_service.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (michal-hudy) [reply components/function-controller/internal/controllers/kubernetes/serviceaccount_service.go:105]: I'm not sure ;)

**Pull request merged**